packages/shared/components/molecules/menu-items/ToggleHiddenAccountMenuItem.svelte

```diff
@@ -1,34 +1,38 @@
 <script lang="ts">
     import { selectedAccount, setNextSelectedAccount } from '@core/account'
     import { localize } from '@core/i18n'
-    import { activeProfile, nonHiddenActiveAccounts, updateActiveAccountMetadata } from '@core/profile'
+    import { activeProfile, nonHiddenActiveAccounts, updateActiveAccountPersistedData } from '@core/profile'
     import { Icon } from '@lib/auxiliary/icon'
-    import { MenuItem } from 'shared/components'
+    import { MenuItem } from '@ui'
 
     export let onClick: () => unknown
 
     function onShowAccountClick(): void {
-        updateActiveAccountMetadata($selectedAccount.index, { hidden: false })
-        onClick && onClick()
+        if ($selectedAccount) {
+            updateActiveAccountPersistedData($selectedAccount.index, { hidden: false })
+            onClick && onClick()
+        }
     }
 
     function onHideAccountClick(): void {
         if ($nonHiddenActiveAccounts.length > 1) {
-            updateActiveAccountMetadata($selectedAccount.index, { hidden: true })
-            if (!$activeProfile.showHiddenAccounts) {
-                setNextSelectedAccount()
+            if ($selectedAccount) {
+                updateActiveAccountPersistedData($selectedAccount.index, { hidden: true })
+                if (!$activeProfile.showHiddenAccounts) {
+                    setNextSelectedAccount()
+                }
+                onClick && onClick()
             }
-            onClick && onClick()
         } else {
             console.error('Not enough accounts visible: ', $nonHiddenActiveAccounts.length)
         }
     }
 </script>
 
 <MenuItem
-    icon={$selectedAccount.hidden ? Icon.View : Icon.Hide}
-    title={localize($selectedAccount.hidden ? 'actions.showAccount' : 'actions.hideAccount')}
-    onClick={() => ($selectedAccount.hidden ? onShowAccountClick() : onHideAccountClick())}
-    disabled={!$selectedAccount.hidden && $nonHiddenActiveAccounts.length <= 1}
+    icon={$selectedAccount?.hidden ? Icon.View : Icon.Hide}
+    title={localize($selectedAccount?.hidden ? 'actions.showAccount' : 'actions.hideAccount')}
+    onClick={() => ($selectedAccount?.hidden ? onShowAccountClick() : onHideAccountClick())}
+    disabled={!$selectedAccount?.hidden && $nonHiddenActiveAccounts.length <= 1}
     {...$$restProps}
 />
```

Comment:
i would rather wrap this component in a 
`{if $selectedAccount}`
so that we know that throughout the whole component this exists

Reply:
updated